p2p/net/swarm/swarm_dial.go

```diff
@@ -44,6 +44,9 @@ var (
 // add loop back in Dial(.)
 const dialAttempts = 1
 
+// dial at most 12 peers concurrently
+const concurrentPeerDials = 12
```

Comment:
we can probably bump the `10` factor to `5`. i worry about putting it lower than that, as otherwise we wont have the chance to prefer local addresses. once we have good estimators, maybe.